sdk/rust/oak/src/handle.rs

```diff
@@ -0,0 +1,102 @@
+include!(concat!(env!("OUT_DIR"), "/oak.handle.rs"));
+
+use crate::OakError;
+
+pub type Handle = oak_abi::Handle;
+
+pub trait HandleVisit {
+    fn visit<F: FnMut(&mut Handle)>(&mut self, visitor: F) -> F;
+}
+
+macro_rules! handle_visit_blanket_impl {
+    ($($t:ty),+) => {
+        $(
+            impl HandleVisit for $t {
+                fn visit<F: FnMut(&mut Handle)>(&mut self, visitor: F) -> F {
+                    visitor
+                }
+            }
+        )+
+    };
+}
+handle_visit_blanket_impl!((), f64, f32, i32, i64, u32, u64, bool, String, Vec<u8>);
+handle_visit_blanket_impl!(
```

Comment:
Is there any reason for splitting these? Either way, please clarify in a comment

Reply:
The first line is all the basic Protobuf/Prost types, the other one is public Oak types that we can't derive the implementation for (and they don't contain handles anyway).